include/morphio/soma.h

```diff
@@ -25,22 +26,33 @@ namespace morphio {
  *
  * @version unstable
  */
-class Soma
+struct Soma
 {
-  public:
+    Soma() = default;
+    Soma(const Soma& soma) = default;
+
+    explicit Soma(const Property::Properties& properties);
+    explicit Soma(const Property::PointLevel& point_properties);
+
     /// Return the  coordinates (x,y,z) of all soma points
-    range<const Point> points() const noexcept {
-        return properties_->_somaLevel._points;
+    std::vector<Point>& points() noexcept {
```

Comment:
So… this makes things mutable, is this still the right place then?

Reply:
Given that there is no readonly/mutable duality with these changes, I thought I might leave it in `morphio/soma.h`. But you are right, it can be moved to mut too.

Reply:
I _suppose_ one could say that the Morphology proper is the (im)mutable entity, in which case the placement of the soma would be fine. But then makes me wonder, shouldn't the section class receive the same treatment?

Reply:
Well, this is an experiment to check if I can pull this through. pybind11 is giving me hell at the time being...
If it will work, yes, I don't see why not do the same for sections and reduce the complexity of the code.

Reply:
Although with sections, the switching from leveraging the ranges to copying the data so that a section data structure is created the same way for both mut/readonly, will result in quite a performance penalty. With soma it doesn't really matter given that it is just one and with very few points.